tests/sentry/snuba/metrics/test_extraction.py

```diff
@@ -460,6 +462,40 @@ def test_spec_wildcard() -> None:
     }
 
 
+@django_db_all
+@pytest.mark.parametrize(
+    "query,title,expected_pattern",
+    [
+        ("title:*[dispatch:*", "backend test [dispatch:something]", r"*\[dispatch:*"),
+        ("title:*{dispatch:*", "test {dispatch:something]", r"*\{dispatch:*"),
+        ("title:*dispatch]:*", "backend dispatch]:", r"*dispatch\]:*"),
+        ("title:*dispatch}:*", "test [dispatch}:", r"*dispatch\}:*"),
+        ("title:*?dispatch*", "backend ?dispatch", r"*\?dispatch*"),
+    ],
+)
+def test_spec_wildcard_escaping(
+    default_project, insta_snapshot, query, title, expected_pattern
+) -> None:
+    spec = OnDemandMetricSpec("count()", query)
+
+    assert spec._metric_type == "c"
+    assert spec.field_to_extract is None
+    assert spec.op == "sum"
+    assert spec.condition == {
+        "name": "event.transaction",
+        "op": "glob",
+        "value": [expected_pattern],
+    }
+
+    # We also validate using Relay's glob implementation to make sure the escaping
+    # is interpreted correctly.
+    assert glob_match(title, expected_pattern, ignorecase=True)
+
+    # We want to validate the json output, to make sure that characters are correctly escaped.
+    metric_spec = spec.to_metric_spec(default_project)
+    insta_snapshot(json.dumps(metric_spec))
```

Comment:
Added snapshot test to validate that serialization to json is happening with the correct escaping.